vertx-web/src/main/java/io/vertx/ext/web/impl/RouteImpl.java

```diff
@@ -269,7 +268,7 @@ synchronized boolean matches(RoutingContext context, String mountPoint, boolean
               String group = m.group(i + 1);
               if(group != null) {
                 final String k = "param" + i;
-                final String value = QueryStringDecoder.decodeComponent(group.replace("+", "%2b"));
```

Comment:
QueryStringDecoder is from Netty. It implements the qery string semantics of the URL and it is correct, however the RFC states that in the path segment (before the question mark) the plus sign should not be translated to a space. Since we use path variables we need a method that does that (Utils.urlDecode()) and it can also replace the netty impl using the boolean flag.